src/Symfony/Component/Translation/Bridge/Lokalise/LokaliseProvider.php

```diff
@@ -45,6 +46,7 @@ public function __construct(HttpClientInterface $client, LoaderInterface $loader
         $this->logger = $logger;
         $this->defaultLocale = $defaultLocale;
         $this->endpoint = $endpoint;
+        $this->icuEnabled = class_exists(\MessageFormatter::class);
```

Comment:
That some useful information. Sounds right for me to use the option, didnt even seen it before. But i think adding the information to the provider would be best at initiation but this is not possible if the re-use the option from the command. So maybe it would also be a solution to able to configure it with the other provider configs?

Reply:
It can be configured for each provider in `translation.yaml`, yes. But be careful, if a provider is configured to use intl-icu format, and `translation:pull` is called without `--intl-icu` option, we have to set the right filename.

Reply:
So what i think is we have 2 options here.

1. Use the `--intl-icu` option from the `TranslationPullCommand` and forward this information to the Providers. 
a) This could happen via an extra argument for the `write` and `read` methods
b) via a configuration method and mutate the state of the service, this one i really dislike

2. Create a new configuration option for the translator component with which the user can then decide if the icu format should be used or not. This could be used on other places too, where symfony right now tries to guess which format is appropiate. This config could be injected into the providers and they can then do whatever they need to enable the icu feature.

Did i miss another option?

The other question i have: Is it right to assume that this is a bug or is it more a feature? Im somewhat not sure at this topic :)

Reply:
I think we must limit to translation providers for now. So, the 2. option seems the right one, but by create the new configuration for each provider:

```yaml

framework:
    default_locale: en
    translator:
        default_path: '%kernel.project_dir%/translations'
        fallbacks:
            - en
        providers:
            lokalise:
                dsn: '%env(CROWDIN_DSN)%'
                intl_icu: true # or false
```

For the question about bug or feature, in case of doubt it's always a feature.

Reply:
I tried to add the configuration for providers. But i added the flag first only for the Lokalise Factory / Provider. Maybe you could have a look again? Thanks!